subworkflows/local/mhc_binding_prediction/main.nf

```diff
@@ -0,0 +1,155 @@
+                        return [meta + [alleles_supported: allele_input_dict['mhcnuggetsii']], file]
+                    netmhcpan: (file.name.contains('netmhcpan_input') && allele_input_dict['netmhcpan'])
+                        return [meta + [alleles_supported: allele_input_dict['netmhcpan']], file]
+                    netmhciipan: (file.name.contains('netmhciipan_input') && allele_input_dict['netmhciipan'])
+                        return [meta + [alleles_supported: allele_input_dict['netmhciipan']], file]
+                    }
+            .set{ ch_prediction_input }
+
+        MHCFLURRY ( ch_prediction_input.mhcflurry )
+        ch_versions = ch_versions.mix(MHCFLURRY.out.versions)
+        ch_binding_predictors_out = ch_binding_predictors_out.mix(MHCFLURRY.out.predicted)
+
+        MHCNUGGETS ( ch_prediction_input.mhcnuggets )
+        ch_versions = ch_versions.mix(MHCNUGGETS.out.versions)
+        ch_binding_predictors_out = ch_binding_predictors_out.mix(MHCNUGGETS.out.predicted)
+
+        MHCNUGGETSII ( ch_prediction_input.mhcnuggetsii )
+        ch_versions = ch_versions.mix(MHCNUGGETSII.out.versions)
+        ch_binding_predictors_out = ch_binding_predictors_out.mix(MHCNUGGETSII.out.predicted)
+
+        if ( "netmhcpan" in tools.tokenize(",") )
+        {
+            NETMHCPAN_IMPORT( parse_netmhc_params("netmhcpan", netmhc_software_meta_json) )
+            NETMHCPAN ( ch_prediction_input.netmhcpan.combine(NETMHCPAN_IMPORT.out.nonfree_tools) )
+            ch_versions = ch_versions.mix(NETMHCPAN.out.versions)
+            ch_binding_predictors_out = ch_binding_predictors_out.mix(NETMHCPAN.out.predicted)
+        }
+
+        if ( "netmhciipan" in tools.tokenize(",") )
+        {
+            NETMHCIIPAN_IMPORT( parse_netmhc_params("netmhciipan", netmhc_software_meta_json) )
+            NETMHCIIPAN ( ch_prediction_input.netmhciipan.combine(NETMHCIIPAN_IMPORT.out.nonfree_tools) )
+            ch_versions = ch_versions.mix(NETMHCIIPAN.out.versions)
+            ch_binding_predictors_out = ch_binding_predictors_out.mix(NETMHCIIPAN.out.predicted)
+        }
+
+    // Join predicted file and subworkflow input file to add inputfile metadata
+    ch_binding_predictors_out
+        .map { meta, file -> [meta.findAll { k, v -> k != 'alleles_supported' }, file] } // drop alleles_supported from meta
+        .groupTuple()
+        .join( ch_peptides_to_predict )
+        .set { ch_binding_predictors_out_meta}
+
+    // Merge predictions from different predictors
+    MERGE_PREDICTIONS( ch_binding_predictors_out_meta )
+    ch_versions = ch_versions.mix(MERGE_PREDICTIONS.out.versions)
+
+    emit:
+    predicted = MERGE_PREDICTIONS.out.merged
+    versions = ch_versions
+}
+
+//==============================================================================
+//                            Auxiliar Functions
+//==============================================================================
+
+// Check if supported tools are specified
```

Comment:
Is there no way to include these function in another file?
I dont know but is the subworkflows/utils_nfcore_epitopeprediction_pipeline not a better way to collect the used functionalities?

Reply:
I asked already in the [#subworkflow nf-core channel](https://nfcore.slack.com/archives/C03D38JNFNJ/p1738141405461969) regarding this. If it should be an nf-core subworkflow (the aim) then it belongs in there